app/models/sales_log.rb

```diff
@@ -111,4 +114,25 @@ def income1_used_for_mortgage?
   def income2_used_for_mortgage?
     inc2mort == 1
   end
+
+  def ppostcode_full=(postcode)
+    if postcode
+      super UKPostcode.parse(postcode).to_s
+    else
+      super nil
+    end
+  end
+
+  def previous_postcode_known?
+    ppcodenk&.zero?
+  end
+
+  def process_postcode(postcode, postcode_known_key, la_inferred_key, la_key)
+    return if postcode.blank?
+
+    self[postcode_known_key] = 0
+    inferred_la = get_inferred_la(postcode)
+    self[la_inferred_key] = inferred_la.present?
+    self[la_key] = inferred_la if inferred_la.present?
+  end
```

Comment:
These are the same as the lettings logs, but I didn't reuse that one because `ppcodenk` == 1 == "Yes", which means that postcode is known in the lettings log if it's set to 1.

In sales log 1 == "No" which meant that the postcode isn't known.

I suspect that the lettings log is wrong and will need to be fixed eventually.

Reply:
Sounds good, sounds like something worth making a separate ticket for